phsb5321/aca.so
# Feature/first working route

## Feature/basic structuring and configuration

This PR implements the following:

- Define models for Person, Book, and Rating using Pydantic.
- Create CRUD functions for Person and Book entities in `usecases.py`. (Note: the functions' logic needs to be implemented using Gremlin)
- Create additional functions for getting books by author, readers by book, and ratings by book.
- Define FastAPI routes for all CRUD operations, as well as additional routes for getting books by author, readers by book, and ratings by book.

This PR sets the foundation for implementing the Gremlin database logic in the `usecases.py` functions to complete the API.

## Testing

To test the API, follow these steps:

1. Clone this repository and navigate to the root directory.
2. Install the required dependencies by running `poetry install`.
3. Start the Gremlin server by running `docker-compose up -d`.
4. Create a new virtual environment and activate it.
5. Start the API by running `uvicorn app.main:app`.
6. Use a tool like `curl`, `httpie`, or `Postman` to make requests to the API endpoints.
7. Verify that the responses match the expected results.
